src/formatDate.js

```diff
@@ -8,7 +8,44 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const slicedDate = date.split(fromFormat[3]);
+  const newSeparator = toFormat[3];
+
+  const dayIndex = fromFormat.indexOf('DD');
+  const monthIndex = fromFormat.indexOf('MM');
+  const yearIndex = Math.abs(dayIndex + monthIndex - 3);
+
+  const newDayIndex = toFormat.indexOf('DD');
+  const newMonthIndex = toFormat.indexOf('MM');
+  const newYearIndex = Math.abs(newDayIndex + newMonthIndex - 3);
+
+  let formatedYear = slicedDate[yearIndex];
+  const newDateParts = [];
+
+  if (
+    fromFormat[yearIndex].length === 4 &&
+    toFormat[newYearIndex].length === 2
+  ) {
+    formatedYear = slicedDate[yearIndex].slice(2);
+  }
+
+  if (
+    fromFormat[yearIndex].length === 2 &&
+    toFormat[newYearIndex].length === 4
+  ) {
+    formatedYear =
+      Number(formatedYear) < 30
+        ? `20${slicedDate[yearIndex]}`
+        : `19${slicedDate[yearIndex]}`;
```

Comment:
The logic for converting from 'YY' to 'YYYY' is correct. However, it's recommended to store the result of the comparison `Number(formattedYear) < 30` in a variable for better readability and to avoid duplicating `slicedDate[yearIndex]`.